src/app/pheno-tool-measure/pheno-tool-measure.component.ts

```diff
@@ -41,24 +41,26 @@ export class PhenoToolMeasureComponent extends StatefulComponent implements OnIn
   public constructor(
     protected store: Store,
     private measuresService: MeasuresService,
-    private datasetsService: DatasetsService,
   ) {
     super(store, PhenoToolMeasureState, 'phenoToolMeasure');
   }
 
   public ngOnInit(): void {
     super.ngOnInit();
 
-    this.dataset = this.datasetsService.getSelectedDataset();
-    if (this.dataset?.phenotypeData) {
-      this.measuresService.getRegressions(this.dataset.id).pipe(take(1)).subscribe(res => {
-        this.regressions = res;
-        this.regressionNames = Object.getOwnPropertyNames(this.regressions);
-      }, () => {
-        // no regressions found in backend
-        // empty error handling block to prevent 404 error showing up in the pheno tool
-      });
-    }
+    this.store.selectOnce((state: { datasetState: DatasetModel}) => state.datasetState).subscribe(state => {
```

Comment:
Convert to switchmap. Use if else scenario: on if continue as usual, on else return of() observable from switchmap and check for undefined at the start of subscribe.